src/components/files/FileBrowser.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+	<template v-if="!items.length">
+		<NoFiles />
+	</template>
+	<template v-else>
+		<div class="flex flex-wrap gap-3">
+			<Folder
+				v-for="folder in folders"
+				:modelValue="folder"
+				@update:modelValue="updateItem(folder)"
+			/>
+		</div>
+		<div class="mt-3 flex flex-wrap gap-3">
+			<File v-for="file in files" :modelValue="file" @update:modelValue="updateItem(file)" />
+		</div>
+	</template>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, type PropType } from 'vue';
+import NoFiles from './NoFiles.vue';
+import Folder from './Folder.vue';
+import File from './File.vue';
+import { FolderClass, FileClass, ItemClass, type FolderType } from '@lib/items';
+import { api } from '@lib/helpers';
+
+const props = defineProps({
+	modelValue: {
+		type: Object as PropType<FolderType>,
+		required: false,
+	},
+});
+
+/**
+ * Items
+ */
+const items = ref<ItemClass[]>([]);
+
+getItems();
+
+function getItems() {
+	fetch(
+		api(
+			`${
+				ItemClass.isItem(props.modelValue) && FolderClass.isFolder(props.modelValue)
```

Comment:
Is this not equivalent to the following?

```suggestion
				props.modelValue
```